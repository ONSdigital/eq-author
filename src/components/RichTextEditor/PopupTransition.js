import React from "react";
import { CSSTransition } from "react-transition-group";
import styled from "styled-components";
import PropTypes from "prop-types";

const Animated = styled.div`
  transform-origin: center center;
  transition: opacity ${props => props.duration / 4}ms ease-in,
    transform ${props => props.duration}ms cubic-bezier(0.175, 0.885, 0.32, 1.4);
  transform: translateY(-3rem) scale(1);

  &.popup-enter {
    opacity: 0.01;
    transform: translateY(0) scale(0.6);
  }

  &.popup-enter-active {
    opacity: 1;
    transform: translateY(-3rem) scale(1);
  }

  &.popup-exit {
    opacity: 1;
    transform: translateY(-3rem) scale(1);
    transition: opacity ${props => props.duration / 4}ms ease-in,
      transform ${props => props.duration}ms ease-in;
  }

  &.popup-exit-active {
    opacity: 0.01;
    transform: translateY(0) scale(0.6);
  }
`;

class PopupTransition extends React.Component {
  static defaultProps = {
    duration: 250,
    in: false
  };

  static propTypes = {
    duration: PropTypes.number,
    children: PropTypes.element
  };

  render() {
    const { duration, children, ...otherProps } = this.props;
    return (
      <CSSTransition timeout={duration} {...otherProps} classNames="popup">
        <Animated duration={duration}>{children}</Animated>
      </CSSTransition>
    );
  }
}

export default PopupTransition;
