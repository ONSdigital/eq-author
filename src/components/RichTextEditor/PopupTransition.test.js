import React from "react";
import { mount } from "enzyme";
import PopupTransition from "./PopupTransition";

describe("components/Popout/PopupTransition", () => {
  let component;

  beforeEach(() => {
    component = mount(
      <PopupTransition duration={200} arbitraryProp="foo">
        <div>hello world</div>
      </PopupTransition>
    );
  });

  it("should render", () => {
    expect(component).toMatchSnapshot();
  });
});
