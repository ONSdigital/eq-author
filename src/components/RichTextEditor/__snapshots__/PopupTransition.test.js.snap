// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Popout/PopupTransition should render 1`] = `
.c0 {
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  -webkit-transition: opacity 50ms ease-in,-webkit-transform 200ms cubic-bezier(0.175,0.885,0.32,1.4);
  -webkit-transition: opacity 50ms ease-in,transform 200ms cubic-bezier(0.175,0.885,0.32,1.4);
  transition: opacity 50ms ease-in,transform 200ms cubic-bezier(0.175,0.885,0.32,1.4);
  -webkit-transform: translateY(-3em) scale(1);
  -ms-transform: translateY(-3em) scale(1);
  transform: translateY(-3em) scale(1);
}

.c0.popup-enter {
  opacity: 0.01;
  -webkit-transform: translateY(0) scale(0.6);
  -ms-transform: translateY(0) scale(0.6);
  transform: translateY(0) scale(0.6);
}

.c0.popup-enter-active {
  opacity: 1;
  -webkit-transform: translateY(-3em) scale(1);
  -ms-transform: translateY(-3em) scale(1);
  transform: translateY(-3em) scale(1);
}

.c0.popup-exit {
  opacity: 1;
  -webkit-transform: translateY(-3em) scale(1);
  -ms-transform: translateY(-3em) scale(1);
  transform: translateY(-3em) scale(1);
  -webkit-transition: opacity 50ms ease-in,-webkit-transform 200ms ease-in;
  -webkit-transition: opacity 50ms ease-in,transform 200ms ease-in;
  transition: opacity 50ms ease-in,transform 200ms ease-in;
}

.c0.popup-exit-active {
  opacity: 0.01;
  -webkit-transform: translateY(0) scale(0.6);
  -ms-transform: translateY(0) scale(0.6);
  transform: translateY(0) scale(0.6);
}

<PopupTransition
  arbitraryProp="foo"
  duration={200}
  in={false}
>
  <CSSTransition
    arbitraryProp="foo"
    classNames="popup"
    in={false}
    timeout={200}
  >
    <Transition
      appear={false}
      arbitraryProp="foo"
      enter={true}
      exit={true}
      in={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={200}
      unmountOnExit={false}
    >
      <PopupTransition__Animated
        arbitraryProp="foo"
        duration={200}
      >
        <div
          className="c0"
        >
          <div>
            hello world
          </div>
        </div>
      </PopupTransition__Animated>
    </Transition>
  </CSSTransition>
</PopupTransition>
`;
